controllers/periodiccurl_controller.go

```diff
@@ -0,0 +1,180 @@
+
+func createPodFromSpec(pcSpec jobsv1alpha1.PeriodicCurlSpec, namespace string, sleepingAmount string) core.Pod {
+	initSleepCommand := "sleep " + sleepingAmount + ";"
+	curlCommand := initSleepCommand + "curl -o /dev/null -s -w '%{time_total}\n' " + pcSpec.Url
+	commands := []string{"/bin/bash", "-c", curlCommand}
+
+	newPod := core.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-pod",
+			Namespace: namespace,
+		},
+		Spec: core.PodSpec{
+			RestartPolicy: "Never",
+			Containers: []core.Container{
+				{
+					Name:    "url-caller",
+					Image:   "quay.io/amirhmi/ubuntu-curl",
+					Command: commands,
+				},
+			},
+		},
+	}
+
+	return newPod
+}
+
+func getPodLogs(pod *core.Pod) string {
+	podLogOpts := core.PodLogOptions{}
+	// use the current context in kubeconfig
+	const KubeConfigPath = "/home/amirhmi/.kube/config"
+	config, err := clientcmd.BuildConfigFromFlags("", KubeConfigPath)
+	if err != nil {
+		panic(err.Error())
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "error in getting access to K8S"
+	}
+	req := clientset.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "error in opening stream"
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "error in copy information from podLogs to buf"
+	}
+	str := buf.String()
+
+	return str
+}
+
+func getChildPod(r *PeriodicCurlReconciler, ctx context.Context, namespace string) (*core.Pod, error) {
+	var childPods core.PodList
+	if err := r.List(ctx, &childPods, client.InNamespace(namespace)); err != nil {
+		return &core.Pod{}, err
+	}
+	if len(childPods.Items) > 0 {
+		return &childPods.Items[0], nil
+	} else {
+		return nil, nil
+	}
+}
+
+func isContainerOfPodSucceeded(pod *core.Pod) bool {
+	const completedContainerMessage = "Completed"
+	for _, c := range pod.Status.ContainerStatuses {
+		if c.State.Terminated != nil && c.State.Terminated.Reason == completedContainerMessage {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *PeriodicCurlReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("periodiccurl", req.NamespacedName)
+
+	var periodicCurl = &jobsv1alpha1.PeriodicCurl{}
+	if err := r.Get(ctx, req.NamespacedName, periodicCurl); err != nil {
+		log.Error(err, "unable to fetch PeriodicCurl")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	childPod, err := getChildPod(r, ctx, req.Namespace)
+	if err != nil {
+		log.Error(err, "unable to get child pod")
+		return ctrl.Result{}, err
+	}
+	if childPod != nil {
+		if isContainerOfPodSucceeded(childPod) {
+			curlResponseTimeLog := getPodLogs(childPod)
+			curlResponseTime, err := strconv.ParseFloat(curlResponseTimeLog[:len(curlResponseTimeLog)-1], 64)
```

Comment:
This line deserves a comment. What is 64?

Reply:
It's for parsing to Float64, and You're right it's confusing and should be constant